api/storage_service.cc

```diff
@@ -900,6 +902,129 @@ void set_storage_service(http_context& ctx, routes& r) {
+                            info.version = sstable->get_version();
+
+                            if (sstable->has_component(sstables::component_type::CompressionInfo)) {
+                                auto& c = sstable->get_compression();
+                                auto cp = sstables::get_sstable_compressor(c);
+
+                                ss::named_maps nm;
+                                nm.group = "compression_parameters";
+                                for (auto& p : cp->options()) {
+                                    ss::map_entry e;
+                                    e.key = p.first;
+                                    e.value = p.second;
+                                    nm.attributes.push(std::move(e));
+                                }
+                                if (!cp->options().count(compression_parameters::SSTABLE_COMPRESSION)) {
+                                    ss::map_entry e;
+                                    e.key = compression_parameters::SSTABLE_COMPRESSION;
+                                    e.value = cp->name();
+                                    nm.attributes.push(std::move(e));
+                                }
+                                info.extended_properties.push(std::move(nm));
+                            }
+
+                            sstables::file_io_extension::attr_value_map map;
+
+                            for (auto* ep : ext.sstable_file_io_extensions()) {
+                                ep->get_attributes(*sstable, map);
+                            }
+
+                            for (auto& p : map) {
+                                struct {
+                                    const sstring& key; 
+                                    ss::sstable& info;
+                                    void operator()(const std::map<sstring, sstring>& map) const {
+                                        ss::named_maps nm;
+                                        nm.group = key;
+                                        for (auto& p : map) {
+                                            ss::map_entry e;
+                                            e.key = p.first;
+                                            e.value = p.second;
+                                            nm.attributes.push(std::move(e));
+                                        }
+                                        info.extended_properties.push(std::move(nm));
+                                    }
+                                    void operator()(const sstring& value) const {
+                                        ss::map_entry e;
+                                        e.key = key;
+                                        e.value = value;
+                                        info.properties.push(std::move(e));                                        
+                                    }
+                                } v{p.first, info};
```

Comment:
Nice visitor trick.

Reply:
We should add "make_override" to some util header...